Source/WebCore/loader/cache/CachedImage.cpp

```diff
@@ -314,6 +318,11 @@ FloatSize CachedImage::imageSizeForRenderer(const RenderElement* renderer, SizeT
     if (!image)
         return { };
 
+#if ENABLE(MULTI_REPRESENTATION_HEIC)
+    if (renderer && renderer->isMultiRepresentationHEIC())
+        return sizeForMultiRepresentationHEIC(renderer->style().fontCascade().primaryFont());
+#endif
```

Comment:
`CachedImage` is just a resource, it knows nothing about layout/rendering.

This also matches the logic below, which uses `imageOrientation()` from the renderer.

I also have the logic here rather than on the renderer / `RenderImageResource`, since `CachedImage::imageSizeForRenderer` is used in many places.